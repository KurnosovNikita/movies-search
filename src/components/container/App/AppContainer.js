import React, {Fragment} from "react";
import FooterContainer from "../Footer/FooterContainer";
import GlobalSearchContainer from "../GlobalSearch/GlobalSearchContainer";
import MovieDetailContainer from "../MovieDetail/MovieDetailContainer";

class AppContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            activePage: 'globalSearch',
            selectedMovie: null,
        };
    }

    render() {
        const {activePage, selectedMovie} = this.state;
        let currentPage;

        switch (activePage) {
            case 'globalSearch':
                currentPage = <GlobalSearchContainer changeToMovieDetailPage={this.changeToMovieDetailPage}/>;
                break;
            case 'movieDetail':
                currentPage = <MovieDetailContainer selectedMovie={selectedMovie}
                                                    changeToMovieDetailPage={this.changeToMovieDetailPage}
                                                    changeToGlobalSearchPage={this.changeToGlobalSearchPage}/>;
                break;
            default:
                break;
        }
        return (
          <Fragment>
              <div className="wrapper">
                  {currentPage}
              </div>
              <FooterContainer footerClassName="footer" contextClassName="footer-content"/>
          </Fragment>
        );
    }

    changeToMovieDetailPage = ({selectedMovie}) => {
        this.setState({selectedMovie: selectedMovie, activePage: 'movieDetail'});
    }

    changeToGlobalSearchPage = () => {
        this.setState({selectedMovie: [], activePage: 'globalSearch'});
    }
}

export default AppContainer;