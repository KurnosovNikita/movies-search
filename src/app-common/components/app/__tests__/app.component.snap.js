import React from 'react';
import {shallow} from 'enzyme';
import App from '../app.component';
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({});

describe('App component', () => {
    let state;
    state = {
        selectedMovie: null,
    };
    describe('with globalSearch component', () => {
        it('should render <App> with <SearchPage>', () => {
            const component = shallow(
              <Provider store={store}>
                  <App {...state}/>
              </Provider>);

            expect(component).toMatchSnapshot();
        });
    });

    describe('with movieDetail component', () => {
        state = {
            selectedMovie: null,
        };

        it('should render <App> with <MovieDetailPage>', () => {
            const component = shallow(
              <Provider store={store}>
                  <App {...state}/>
              </Provider>);

            component.update();
            expect(component).toMatchSnapshot();
        });
    });
});
