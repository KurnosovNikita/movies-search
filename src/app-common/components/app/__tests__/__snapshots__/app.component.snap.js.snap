// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component with globalSearch component should render <App> with <SearchPage> 1`] = `
<Base
  data={
    <Search
      changeToMovieDetailPage={[Function]}
    />
  }
/>
`;

exports[`App component with movieDetail component should render <App> with <MovieDetailPage> 1`] = `
<Base
  data={
    <Search
      changeToMovieDetailPage={[Function]}
    />
  }
/>
`;
